### PR TITLE
Semi-manually update all 'Starting with VERSION' Obsolete attributes to use 'ObsoletedOSPlatform'

### PR DESCRIPTION
**Pull request by chamons**
- Part of https://github.com/xamarin/xamarin-macios/issues/15849
- These were done with a few vim macros and some manual edits, so mistakes are possibly but new typos are less likely.

### CODE REVIEW

**Review by tj-devel709 — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

File: src/AppKit/NSFont.cs [hunk 1]
```diff
@@ -174,9 +172,7 @@ public static NSFont ControlContentFontOfSize (nfloat fontSize)
 #if NET
 		[SupportedOSPlatform ("macos")]
 		[UnsupportedOSPlatform ("macos10.13")]
-#if MONOMAC
-		[Obsolete ("Starting with macos10.13.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
-#endif
+		[ObsoletedOSPlatform ("macos10.13.")]
```
Comment (rolfbjarne) [inline src/AppKit/NSFont.cs:175]: This is probably an invalid version number (and this happens in a few more places in this PR as well).

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: src/AppKit/NSFont.cs (hunk 1 above)
Comment (chamons) [reply src/AppKit/NSFont.cs:175]: Arg, I thought I got them all (there was a bug in my macro).

I'll find them all shortly.

**Review by rolfbjarne — APPROVED** [review]

File: src/AppKit/NSFont.cs (hunk 1 above)
Comment (dalexsoto) [reply src/AppKit/NSFont.cs]: woah nice catch 👀

**Comment by rolfbjarne** [comment]
Test failures are due to ##15939, so this looks good to go.

**Pull request merged**